### ISSUE TITLE
Commercetools\Core\Model\Common\Collection::offsetUnset() forgets to remove the Key

### ISSUE THREAD

**Issue by b3nl** [commercetools/commercetools-php-sdk#306]
If you remove something from the collection thru the array access the key is forgotten ... but is still used in the iterator.

```
public function offsetUnset($offset)
    {
        unset($this->rawData[$offset]);
        unset($this->typeData[$offset]);
    }
```

**Comment by jayS-de** [comment]
Please see #307. If you want to try if the issue is fixed you could use dev-collection_fix as version constraint for the SDK

I will backport the fix also to version 1.7.1

**Comment by jayS-de** [comment]
The fix is now also backported to the 1.7 branch. You could try it with: 1.7.x-dev
